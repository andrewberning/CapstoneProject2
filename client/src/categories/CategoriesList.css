.CategoriesList {
  margin: 50px;
}

.CategoriesList .title {
  text-align: center;
}