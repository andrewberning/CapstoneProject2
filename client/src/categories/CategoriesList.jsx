import { useState, useEffect } from "react";
import ShoplyApi from "../api/api";
import CategoriesCardList from "./CategoriesCardList";

export default function CategoriesList() {
  const [categories, setCategories] = useState([]);

  
  useEffect(function getAllCategoriesOnMount() {
    async function getAllCategories() {
      let categories = await ShoplyApi.getCategories();
      setCategories(categories);
    }

    getAllCategories();
  }, [])

  return (
    <div className="ClothingList">
      <h2>Categories</h2>
      <CategoriesCardList categories={categories} />
    </div>
  )
}