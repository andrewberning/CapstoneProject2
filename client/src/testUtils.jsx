import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
