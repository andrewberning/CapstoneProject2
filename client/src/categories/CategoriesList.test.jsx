import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CategoriesList from "../categories/CategoriesList";
import { UserProvider } from "../testUtils";

describe('CategoriesList component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <CategoriesList />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <CategoriesList />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});