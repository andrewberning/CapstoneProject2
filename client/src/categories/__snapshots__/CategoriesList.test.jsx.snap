// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoriesList component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CategoriesList"
  >
    <h2
      class="title"
    >
      Categories
    </h2>
    <div
      class="CategoriesCardList"
    >
      <ul
        class="categories-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
